### PR TITLE
[flytepropeller] Add test under flytepropeller/manager

### LINKED ISSUE #4209: [Housekeeping] Improve Test Coverage
### Describe the issue

The test coverage of some components is low. We should add more tests to Flyte repo.
Creating multiple PRs to fix it is fine since it cannot be done in one PR.

image

### What if we do not do this?

Test Coverage > 80%

### Related component(s)

https://app.codecov.io/gh/flyteorg/flyte

### Are you sure this issue hasn't been raised already?

- [X] Yes

### Have you read the Code of Conduct?

- [X] Yes

### PR DESCRIPTION
**Pull request by chaohengstudent**
## Tracking issue

Fixes #4209 

## Describe your changes

Add tests in flytepropeller/manager to leverage coverage.
1. manager_test.go
2. ./config/config_test.go
3. ./shardstrategy/enviroment_test.go
4. ./shardstrategy/enviroment_hash_test.go
5. ./shardstrategy/shared_strategy_test.go

If there are any improvements for these tests please let me know.

## Check all the applicable boxes 

- [ ] I updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] All commits are signed-off.

## Screenshots

from:
image
to:
image

### CODE REVIEW

File: flytepropeller/manager/shardstrategy/environment_test.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
+package shardstrategy
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestEnvironmentHashCode(t *testing.T) {
+	strategy1 := &EnvironmentShardStrategy{
```
Comment (pingsutw) [inline flytepropeller/manager/shardstrategy/environment_test.go:10]: Nit: could we rename it?
`strategy1` -> `expectedStrategy`
`strategy2` -> `actualStrategy`

File: flytepropeller/manager/shardstrategy/hash_test.go [hunk 2]
```diff
@@ -23,3 +23,17 @@ func TestComputeKeyRange(t *testing.T) {
 		assert.Equal(t, keyspaceSize, keysCovered)
 	}
 }
+
+func TestHashHashCode(t *testing.T) {
+	strategy1 := &HashShardStrategy{
```
Comment (pingsutw) [inline flytepropeller/manager/shardstrategy/hash_test.go:28]: ditto

File: flytepropeller/manager/shardstrategy/environment_test.go (hunk 1 above)
Comment (chaohengstudent) [reply flytepropeller/manager/shardstrategy/environment_test.go]: sure!

**Review by pingsutw — COMMENTED** [review]
could you resolve the merge conflict

**Comment by chaohengstudent** [comment]
> could you resolve the merge conflict

sure. have resolved that.

**Review by pingsutw — APPROVED** [review]

**Pull request merged**